components/automate-gateway/gateway/middleware/auth_interceptor_internal_test.go

```diff
@@ -74,6 +102,38 @@ func TestHeaderAuthValidForClientAndPeer(t *testing.T) {
 	}
 }
 
+func TestComboAuthForwardsMetadataCorrectly(t *testing.T) {
```

Comment:
I'd like to see a failure case added here, where authz-service returns an unauthorized error ([here](https://github.com/chef/automate/blob/master/components/automate-gateway/gateway/middleware/authv2/authv2.go#L112)) and it bubbles that error up correctly.